calculator.go

```diff
@@ -1,13 +1,86 @@
 // Package calculator provides a library for simple calculations in Go.
 package calculator
 
+import (
+	"errors"
+	"fmt"
+	"go/token"
+	"go/types"
+	"math"
+	"strconv"
+)
+
 // Add takes two numbers and returns the result of adding them together.
-func Add(a, b float64) float64 {
-return a + b
+func Add(n1, n2 float64, extra ...float64) float64 {
+	var total float64 = n1 + n2
+	for i := range extra {
+		total += extra[i]
+	}
+	return total
 }
 
 // Subtract takes two numbers and returns the result of subtracting the second
 // from the first.
-func Subtract(a, b float64) float64 {
-	return b - a
+func Subtract(n1, n2 float64, extra []float64) float64 {
+	result := n1 - n2
+	for i := range extra {
+		result -= extra[i]
+	}
+	return result
+}
+
+// Multiply takes two numbers and returns the result of multiplication them together.
+func Multiply(n1, n2 float64, extra []float64) float64 {
+	result := n1 * n2
+	for i := range extra {
+		result *= extra[i]
+	}
+	return result
+}
+
+// Divide takes two numbers and returns the result of divion and an error message
+func Divide(n1, n2 float64, extra []float64) (float64, error) {
+	var divideByZeroError = "Cannot divide by zero"
+	if n2 == 0 {
+		return 0, errors.New(divideByZeroError)
+	}
+	result := n1 / n2
+	for i := range extra {
```

Comment:
If we want the _elements_ from the slice, as opposed to the indexes, we can use the second value returned by `range`:

```go
for _, n := range extra {
	if n == 0 { 
...
```